### PR TITLE
Secrets fixups

### PR DESCRIPTION
**Pull request by thockin**
Things I wanted to fix as I reviewed all the secrets work.

### CODE REVIEW

**Comment by markturansky** [comment]
Will there be any attempt to extract an interface from these or otherwise make common attributes across all types?  I am referring to the comment you left in another thread where you might want to request an EmptyDir with specific Throughput performance characteristics.

**Comment by vmarmol** [comment]
Assigning to @erictune, feel free to rebalance.

**Comment by thockin** [comment]
There may be common interfaces across them, but I want to use NFS and iscsi
as drivers for that, rather than just guessing.

**Comment by erictune** [comment]
LGTM.  Not passing tests.

**Comment by erictune** [comment]
looks like there is a new precommit hook.  maybe a rebase will help

**Comment by thockin** [comment]
rebased and pushed.

File: pkg/api/types.go [hunk 1]
```diff
@@ -1329,7 +1329,7 @@ const MaxSecretSize = 1 * 1024 * 1024
 type SecretType string
```
Comment (pmorie) [inline pkg/api/types.go]: @bgrant0607 wanted this renamed to `SecretKind`

**Comment by pmorie** [comment]
@derekwaynecarr FYI

**Comment by pmorie** [comment]
@thockin LGTM with the exception of renaming `SecretType` to `SecretKind`.

**Comment by pmorie** [comment]
@thockin Rereading the comment on #4514 to make sure all feedback was accounted for here, I found this comment https://github.com/GoogleCloudPlatform/kubernetes/pull/4514#discussion_r25120054.  Tl;dr: need description on fields.

File: pkg/api/types.go (hunk 1 above)
Comment (thockin) [inline pkg/api/types.go]: Will do in next push

**Comment by thockin** [comment]
Pushed update with SecretKind.  It looks like descriptions are OK now - I don't find any missing.

**Comment by pmorie** [comment]
@thockin One last nit :) @bgrant0607 wanted `SecretKind` to appear before `Data`.

**Comment by thockin** [comment]
Little problem: if we rename Secret.Type to Secret.Kind we now conflict with TypeMeta.Kind, which is embedded in Secret.  If we rename it to SecretKind then it is different from a dozen other "Kind" fields, which just happen to not be on API objects.

@bgrant0607 for an opinion on what to do with this.

**Comment by bgrant0607** [comment]
Annoyingly, this is only a problem in the Go inline structs, not in the json.  Taking a look.

**Comment by thockin** [comment]
Propose to drop the Type/Kind from this series and deal with it on its own?  If OK, I'll push a new update.

File: docs/design/secrets.md [hunk 2]
```diff
@@ -277,15 +277,15 @@ type Secret struct {
...
 }
 
-type SecretType string
+type SecretKind string
 
 const (
-    SecretTypeOpaque              SecretType = "opaque"           // Opaque (arbitrary data; default)
-    SecretTypeKubernetesAuthToken SecretType = "kubernetes-auth"  // Kubernetes auth token
-    SecretTypeDockerRegistryAuth  SecretType = "docker-reg-auth"  // Docker registry auth
+    SecretKindOpaque              SecretKind = "opaque"           // Opaque (arbitrary data; default)
```
Comment (bgrant0607) [inline docs/design/secrets.md]: These don't follow our constant convention because?

**Comment by thockin** [comment]
Also we have AffinityType floating around

**Comment by bgrant0607** [comment]
And LimitType, FSType, CapabilityType, CauseType.

Kinds include only PodConditionKind and NodeConditionKind.

So I guess it's the Kinds that are odd.

If you want to revert to SecretType, I'm ok with that.

**Comment by thockin** [comment]
I think this latest push addresses all comments except Kind/Type which we can reconsider on its own.

**Comment by pmorie** [comment]
@thockin LGTM :+1:

Thanks for fixing these nits.

**Comment by thockin** [comment]
LGTM accomplished.  Will self-commit when green.  Will think on type/kind.

**Pull request merged**